lib/tests/test_git.rs

```diff
@@ -2166,3 +2167,41 @@ fn create_rooted_commit<'repo>(
         .set_author(signature.clone())
         .set_committer(signature)
 }
+
+#[test]
+fn test_parse_gitmodules() {
+    let temp_dir = testutils::new_temp_dir();
+    let mut temp_file = NamedTempFile::new_in(&temp_dir).unwrap();
+    temp_file
+        .write_all(
+            r#"
+[submodule "wellformed"]
+url = https://github.com/martinvonz/jj
+path = mod
+update = checkout # Extraneous config
```

Comment:
I'll need to check if that last one is well-formed or not. Git probably considers it valid but it might not have been a reasoned decision.

Reply:
Good call! looks like the defined behavior is to skip the second one https://github.com/git/git/blob/master/submodule-config.c#L398, which I found counterintuitive because Git usually checks last-one-wins.